src/main/java/org/ozwillo/dcexporter/service/ResourceTransformerService.java

```diff
@@ -92,7 +92,7 @@ public String resourcesToJson(List<DCResource> resources, List<String> excludedF
 
         resources.forEach(resource -> {
             Map<String,DCResource.Value> values = resource.getValues().entrySet().stream()
-                    .filter(entry -> excludedFields.contains(entry.getKey()))
+                    .filter(entry -> excludedFields == null || !excludedFields.contains(entry.getKey()))
```

Comment:
```suggestion
                    .filter(entry -> excludedFields == null || excludedFields.contains(entry.getKey()))
```

Reply:
`Stream#filter` is a _filter in_, not _filter out_; it's about what you want to **keep**, not what you want to exclude:

> Returns a stream consisting of the elements of this stream that **match** the given predicate.

[*Source*](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#filter-java.util.function.Predicate-) (emphasis mine)

So I'd say @bilani's patch is right, and your suggestion wrong (fwiw, if it worked like you thought it did, I'd haver rather used `excludedFields != null && excludedFields.contains(…)`, otherwise you would have excluded everything if `excludedFields` was `null`)

Reply:
@tbroyer right ! thank's for the explanations